scripts/check-next-viewport-boundaries.sh

```
@@ -0,0 +1,66 @@
+# extensions/cornerstone/src/services/ViewportService/backends/README.md).
+#
+# Enforces:
+#   1. csUtils.isGenericViewport is called ONLY from the adapter dispatch module.
+#   2. isNextViewportsEnabled() is called ONLY from the sanctioned flag-read list.
+#   3. CornerstoneViewportService and the UI layer (hooks/, Viewport/, components/)
+#      contain no per-lane branching at all.
+#
+# Run from the repo root: ./scripts/check-next-viewport-boundaries.sh
+set -u
+
+cd "$(dirname "$0")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)